### PR TITLE
Built chardet for RHEL7.7 & RHEL8

### PR DESCRIPTION
**Pull request by myersCody**
# chardet
```
make add-scratch-repo url=http://brew-task-repos.usersys.redhat.com/repos/scratch/cmyers/python-chardet/3.0.4/9.el7d/python-chardet-3.0.4-9.el7d-scratch.repo
```
## RHEL 8 Tests:
```
make add-scratch-repo url=http://brew-task-repos.usersys.redhat.com/repos/scratch/cmyers/python-chardet/3.0.4/9.el8d/python-chardet-3.0.4-9.el8d-scratch.repo
```

### CODE REVIEW

File: rcm_data/deps/chardet/Makefile [hunk 1]
```diff
@@ -3,65 +3,152 @@ PYTHON		= $(shell which python)
...
+	cd /etc/yum.repos.d/; wget -O - $(url) >> $(notdir $(url))
+
+test-scratch:
+	yum -y install python3-chardet
+	@echo "######################################################################"
+	cat test_chardet.py
+	@echo "######################################################################"
+	python3 test_chardet.py
+
+test-scratch-7: test-scratch
```
Comment (abaiken) [inline rcm_data/deps/chardet/Makefile]: add both `test-scratch-7/test-scratch-8` to the make help

File: rcm_data/deps/chardet/Makefile [hunk 2]
```diff
@@ -3,65 +3,152 @@ PYTHON		= $(shell which python)
...
+test-scratch-8: test_scratch
```
Comment (abaiken) [inline rcm_data/deps/chardet/Makefile]: This doesn't work. `test_scratch` needs to be `test-scratch`

File: rcm_data/deps/chardet/test_chardet.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import urllib
+import chardet
+import urllib.request
+with urllib.request.urlopen("http://www.python.org") as url:
+	rawdata = url.read()
+	chardet.detect(rawdata)
+#should see: {'encoding': 'ISO-8859-1', 'confidence': 0.73, 'language': ''}
```
Comment (abaiken) [inline rcm_data/deps/chardet/test_chardet.py]: This comment leads me to believe I should see the above, but there is no output from running this file: 
```
Complete!
######################################################################
cat test_chardet.py
import urllib
import chardet
import urllib.request
with urllib.request.urlopen("http://www.python.org") as url:
        rawdata = url.read()
        chardet.detect(rawdata)
#should see: {'encoding': 'ISO-8859-1', 'confidence': 0.73, 'language': ''}
######################################################################
python3 test_chardet.py
[root@rhel8 chardet]# python3 test_chardet.py 
[root@rhel8 chardet]# 
```

**Comment by myersCody** [comment]
chardet demo: https://drive.google.com/open?id=1jN3FRajempETft6i0zG_Sug4Gl86usSA

File: rcm_data/deps/chardet/Makefile [hunk 4]
```diff
@@ -3,65 +3,153 @@ PYTHON		= $(shell which python)
...
+# anytime the timestamp on the file changes.
+
+init-test-7:
+	cd ../../config/; rpm -i redhat-internal-cert-install-0.1-9.el7.csb.noarch.rpm | true
+	cd ../../config/; cp -f ./*.repo /etc/yum.repos.d/
+	rm /etc/yum.repos.d/rcm-tools-rhel-8-baseos.repo
+	yum -y install python36
+	yum remove python-chardet
+
+init-test-8:
```
Comment (kholdaway) [inline rcm_data/deps/chardet/Makefile]: This command fails for me.  @abaiken is trying now.

```[root@vrhel8test chardet]# make init-test-8
cd ../../config/; cp -f ./rcm-tools-rhel-8-baseos.repo /etc/yum.repos.d/
yum -y install python3
Updating Subscription Management repositories.
Unable to read consumer identity
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
RCM Tools for Red Hat Enterprise Linux 8 BaseOS (RPMs)                                          0.0  B/s |   0  B     00:01
RCM Tools for Red Hat Enterprise Linux 8 appstream                                              3.2 kB/s | 3.2 kB     00:01
RCM Tools for Red Hat Enterprise Linux 8 BaseOS - Optional (RPMs)                               0.0  B/s |   0  B     00:01
Failed to synchronize cache for repo 'rcm-tools-rhel-8-baseos-rpms', ignoring this repo.
Failed to synchronize cache for repo 'rcm-tools-rhel-8-server-optional-rpms', ignoring this repo.
Error:
 Problem: conflicting requests
  - nothing provides python3-setuptools needed by python36-3.6.8-1.module+el8.1.0+2993+d789589b.x86_64
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
make: *** [Makefile:142: init-test-8] Error 1```

**Comment by kholdaway** [comment]
works.  it was me.

File: Vagrantfile [hunk 5]
```diff
@@ -22,7 +22,7 @@ Vagrant.configure("2") do |config|
 
   config.vm.define "vrhel8test" do |rhel8test|
     rhel8test.vm.box = "generic/rhel8"
-    #rhel8test.vm.hostname = "vrhel8test"
+    rhel8test.vm.hostname = "vrhel8test"
```
Comment (myersCody) [inline Vagrantfile]: This change has broken my vagrant machine.

File: rcm_data/deps/chardet/Makefile (hunk 4 above)
Comment (kholdaway) [reply rcm_data/deps/chardet/Makefile]: Works 😅 Sorry about the false alarm

File: Vagrantfile (hunk 5 above)
Comment (myersCody) [reply Vagrantfile]: Virtualbox version needed to be: 5.26

File: rcm_data/deps/chardet/Makefile (hunk 4 above)
Comment (myersCody) [reply rcm_data/deps/chardet/Makefile]: Required new secrets.tar.gz

**Pull request merged**